streams/src/test/java/org/apache/kafka/streams/state/internals/CompositeReadOnlySessionStoreTest.java

```diff
@@ -113,34 +113,35 @@ public void shouldThrowInvalidStateStoreExceptionOnRebalance() throws Exception
     }
 
     @Test
-    public void shouldThrowInvalidStateStoreExceptionIfFetchThrows() {
+    public void shouldThrowInvalidStateStoreExceptionIfUnderlyingSessionStoreFetchThrows() {
+        underlyingSessionStore.setOpen(false);
         try {
-            underlyingSessionStore.setOpen(false);
             underlyingSessionStore.fetch("key");
             fail("Should have thrown InvalidStateStoreException with underlying session store");
         } catch (InvalidStateStoreException e) { }
+    }
 
+    @Test
+    public void shouldThrowInvalidStateStoreExceptionIfSessionFetchThrows() {
+        underlyingSessionStore.setOpen(false);
         try {
             sessionStore.fetch("key");
             fail("Should have thrown InvalidStateStoreException with session store");
         } catch (InvalidStateStoreException e) { }
     }
 
-    @Test
-    public void shouldThrowNullPointerException() {
-        try {
-            sessionStore.fetch(null);
-            fail("Should have thrown NullPointerException while fetching null key");
-        } catch (NullPointerException e) { }
+    @Test(expected = NullPointerException.class)
+    public void shouldThrowNullPointerExceptionIfFetchingNullKey() {
+        sessionStore.fetch(null);
+    }
 
-        try {
-            underlyingSessionStore.put(new Windowed<>((String) null, new SessionWindow(0, 0)), 1L);
-            fail("Should have thrown NullPointerException while putting null window key");
-        } catch (NullPointerException e) { }
+    @Test(expected = NullPointerException.class)
+    public void shouldThrowNullPointerExceptionIfPuttingNullSessionWindowKey() {
+        underlyingSessionStore.put(new Windowed<>((String) null, new SessionWindow(0, 0)), 1L);
```

Comment:
It's unclear here, that `put` throws and not `new Windowed()` or `new SeeesionWindow()` -- should split and use try-fail-catch pattern here.